src/main/java/seedu/stockpal/commands/NewCommand.java

```diff
@@ -3,25 +3,30 @@
 import seedu.stockpal.data.ProductList;
 import seedu.stockpal.data.product.Product;
 
+import static seedu.stockpal.ui.Ui.printToScreen;
+
 
 public class NewCommand extends ListActionCommand {
     public static final String COMMAND_KEYWORD = "new";
     public static final String COMMAND_USAGE = COMMAND_KEYWORD + ": ";
-    private static Integer pid = 1;
     protected ProductList productList;
     private final Product toAdd;
 
 
-
-
     public NewCommand(ProductList productList, String name, Integer quantity, Double price, String description) {
-        this.toAdd = new Product(name, quantity, price, description, pid++);
+        int pid;
+        if (productList.isEmpty()) {
+            pid = 1;
+        } else {
+            pid = productList.getSize() + 1;

```

Comment:
Temporal fix, need to consider potential Pid overlap in the case that a product gets deleted

Reply:
Considered product gets deleted case, fixed this issue.